src/argilla/training/base.py

```diff
@@ -0,0 +1,216 @@
+                self._multi_label = True
+        elif isinstance(self.rg_dataset_snapshot, rg.DatasetForTokenClassification):
+            self._rg_dataset_type = rg.DatasetForTokenClassification
+            self._required_fields = ["id", "text", "tokens", "annotation", "ner_tags"]
+        elif isinstance(self.rg_dataset_snapshot, rg.DatasetForText2Text):
+            self._rg_dataset_type = rg.DatasetForText2Text
+            self._required_fields = ["id", "text", "annotation"]
+        else:
+            raise NotImplementedError(f"Dataset type {type(self.rg_dataset_snapshot)} is not supported.")
+
+        self.dataset_full = rg.load(name=self._name, fields=self._required_fields, **load_kwargs)
+
+        framework = Framework(framework)
+        if framework is Framework.SPACY:
+            import spacy
+
+            self.dataset_full_prepared = self.dataset_full.prepare_for_training(
+                framework=framework,
+                train_size=self._train_size,
+                seed=self._seed,
+                lang=spacy.blank("en") if lang is None else lang,
+            )
+        else:
+            self.dataset_full_prepared = self.dataset_full.prepare_for_training(
+                framework=framework,
+                train_size=self._train_size,
+                seed=self._seed,
+            )
+
+        if framework is Framework.SETFIT:
+            if self._rg_dataset_type != rg.DatasetForTextClassification():
+                raise NotImplementedError(f"{Framework.SETFIT} only support `TextClassification` tasks.")
+            self._trainer = ArgillaSetFitTrainer(
+                record_class=self._rg_dataset_type._RECORD_TYPE,
+                dataset=self.dataset_full_prepared,
+                multi_label=self._multi_label,
+                seed=self._seed,
+                model=self.model,
+            )
+        elif framework is Framework.TRANSFORMERS:
+            if self._rg_dataset_type not in [rg.DatasetForTokenClassification, rg.DatasetForTextClassification]:
+                raise NotImplementedError("`argilla.training` does not support `Text2Text` tasks yet.")
+            self._trainer = ArgillaTransformersTrainer(
+                record_class=self._rg_dataset_type._RECORD_TYPE,
+                dataset=self.dataset_full_prepared,
+                multi_label=self._multi_label,
+                seed=self._seed,
+                model=self.model,
+            )
+        elif framework is Framework.SPACY:
+            if self._rg_dataset_type == rg.DatasetForText2Text:
+                raise NotImplementedError("`argilla.training` does not support `Text2Text` tasks yet.")
+            self._trainer = ArgillaSpaCyTrainer(
+                record_class=self._rg_dataset_type._RECORD_TYPE,
+                dataset=self.dataset_full_prepared,
+                model=self.model,
+                multi_label=self._multi_label,
+                seed=self._seed,
+            )
```

Comment:
Also, this is duplicated 😩 I assume that's my bad due to the merge of my branch with this one.

```suggestion
```

Reply:
Oops my bad, I thought that the `ArgillaSpaCyTrainer` was duplicated but it's not!